### ISSUE TITLE
Confusing error with type completeness check

### ISSUE THREAD

**Issue by RobertCraigie** [microsoft/pyright#2637]
**Describe the bug**

`pkg/foo.py`
```py
from typing import Callable, Any

AnyCallable = Callable[..., Any]

def validator(field: str) -> Callable[[AnyCallable], classmethod]:
    ...

class Foo:
    name: str

    @validator('name')
    def validate_name(cls, value: str) -> str:
        return value
```

**To Reproduce**

`pyright --verifytypes pkg`

```
Public modules: 1
   pkg.foo

Symbols used in public interface:
prisma.foo.Foo.validate_name
  pkg/foo.py:16:9 - error: Type argument 0 has unknown type
```

**Expected behavior**

The error message should indicate that the unknown type results from the incorrectly typed decorator.

**VS Code extension or command-line**

command-line pyright 1.1.191

**Comment by erictraut** [comment]
I've improved the error messages. The change will be included in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.192, which I just published. It will also be included in the next release of pylance.
